src/Composer/Satis/PackageSelection/PackageSelection.php

```diff
@@ -126,7 +151,16 @@ public function select(Composer $composer, $verbose)
 
         $repos = $composer->getRepositoryManager()->getRepositories();
         $pool = new Pool($this->minimumStability);
-        foreach ($repos as $repo) {
+        foreach ($repos as $key => $repo) {
+            if ($this->hasRepositoryFilter()) {
+                $repoConfig = $repo->getRepoConfig();
```

Comment:
@realshadow Don't worry about the delay. I'm happy that we'll be able to speed up our satis build soon. Thank you for your work.